classes/class-wc-connect-shipping-label.php

```diff
@@ -36,89 +36,111 @@ protected function get_form_data( WC_Order $order ) {
+			foreach( WC()->countries->get_countries() as $code => $name ) {
+				$result[ $code ] = array( 'name' => html_entity_decode( $name ) );
 			}
-
-			return $formData;
+			foreach( WC()->countries->get_states() as $country => $states ) {
+				$result[ $country ][ 'states' ] = array();
+				foreach ( $states as $code => $name ) {
+					$result[ $country ][ 'states' ][ $code ] = html_entity_decode( $name );
+				}
+			}
+			return $result;
 		}
 
 		protected function get_form_layout() {
 			$layout = array();
 
 			$address_fields = array(
 				array(
-					'key' => 'first_name',
-					'validation_hint' => 'This field is required.',
-				),
-				array(
-					'key' => 'last_name',
+					'key' => 'name',
+					'validation_hint' => __( 'Required.', 'woocommerce' ),
 				),
 				array(
 					'key' => 'company',
 				),
 				array(
 					'key' => 'address_1',
-					'validation_hint' => 'This field is required.',
+					'validation_hint' => __( 'Required.', 'woocommerce' ),
 				),
 				array(
 					'key' => 'address_2',
 				),
 				array(
 					'key' => 'city',
-					'validation_hint' => 'This field is required.',
+					'validation_hint' => __( 'Required.', 'woocommerce' ),
 				),
 				array(
 					'key' => 'state',
-					'validation_hint' => 'This field is required.',
+					'type' => 'state',
+					'validation_hint' => __( 'Required.', 'woocommerce' ),
 				),
 				array(
 					'key' => 'postcode',
-					'validation_hint' => 'This field is required.',
+					'validation_hint' => __( 'Required.', 'woocommerce' ),
 				),
 				array(
 					'key' => 'country',
-					'validation_hint' => 'This field is required.',
+					'validation_hint' => __( 'Required.', 'woocommerce' ),
+					'type' => 'country',
 				),
 			);
-			$address_summary = '{first_name} {last_name}\\n{address_1} {address_2}\\n{city}, {postcode} {state}, {country}';
+			$address_summary = '{name}\\n{address_1} {address_2}\\n{city}, {postcode} {state}, {country}';
 
 			$orig_address_fields = $address_fields;
 			$dest_address_fields = $address_fields;
 			foreach( $address_fields as $index => $_ ) {
 				$orig_address_fields[ $index ][ 'key' ] = 'orig_' . $address_fields[ $index ][ 'key' ];
 				$dest_address_fields[ $index ][ 'key' ] = 'dest_' . $address_fields[ $index ][ 'key' ];
+				if ( 'state' === $address_fields[ $index ][ 'key' ] ) {
+					$orig_address_fields[ $index ][ 'country_field' ] = 'orig_country';
+					$dest_address_fields[ $index ][ 'country_field' ] = 'dest_country';
+				}
 			}
 
 			$layout[] = array(
 				'type' => 'step',
-				'tab_title' => 'Origin',
-				'title' => 'Generate shipping label: Address verification (origin)',
-				'description' => 'asdfghjkjhgfdsaASDFGHJJHGFDSA',
+				'tab_title' => __( 'Origin', 'woocommerce' ),
+				'title' => __( 'Generate shipping label: Origin address', 'woocommerce' ),
+				'description' => __( 'Confirm the address of the recipient is correct to ensure the package will get to the correct destination', 'woocommerce' ),
```

Comment:
I think this should say something like "Confirm the address you are shipping from is correct..." right?